tasks/rsync.py

```diff
@@ -0,0 +1,218 @@
+        self.waitTime = self.config.get('waittime', 5)
+
+        self.fs = Filesystem(self.ctx)
+
+        self.snapEnable = bool(self.config.get('snapenable', False))
+        self.logger.info('Snapenable param: {}'.format(self.snapEnable))
+
+        if self.config.get('data_dir'):
+            self.workUnit = True
+            self.dataDir = misc.get_testdir(self.ctx)
+            self.sourceDir = self.dataDir + '/workunit.{}/'.format(self.config.get('data_dir'))
+        else:
+            self.snapShot = None
+            self.snapDummy = None
+            self.iteration = 0
+            self.dataDir = 'source'
+            self.sourceDir = self.dataDir + '/subdir'
+
+    def _run(self):
+        try:
+            self.do_rsync()
+        except:
+            # Log exceptions here so we get the full backtrace (it's lost
+            # by the time someone does a .get() on this greenlet)
+            self.logger.exception("Exception in do_rsync:")
+            raise
+
+    def log(self, x):
+        """Write data to logger assigned to this MDThrasher"""
+        self.logger.info(x)
+
+    def stop(self):
+        time.sleep(self.runTime)
+        self.stopping.set()
+
+    def rsync_io(self):
+        try:
+            if self.snapEnable:
+                # Rsync snapshot data to rsync directory
+                self.my_mnt.run_shell(["rsync", "-azvh", "{}".format(self.snapShot), "rsyncdir/dir1/"])
+            else:
+                # Rsync data from data directory to rsync directory
+                self.my_mnt.run_shell(["rsync", "-azvh", "{}".format(self.sourceDir), "rsyncdir/dir1/"])
+
+        except CommandFailedError:
+            self.logger.error('rsync command failed asserting rsync')
+            raise
+
+    def do_rsync(self):
+
+        # Get mount objects
+        mounts = [v for k, v in sorted(self.ctx.mounts.items(), lambda a, b: cmp(a[0], b[0]))]
+
+        # Using first mount point of cephfs for run rsync test.
+        if len(mounts):
+            self.my_mnt = mounts.pop(0)
+        else:
+            assert len(mounts) > 0, self.logger.error('No mount available asserting rsync')
+
+        if self.snapEnable:
+            # Enable snapshots
+            self.fs.mon_manager.raw_cluster_cmd("mds", "set", "allow_new_snaps", "true", "--yes-i-really-mean-it")
+            self.snapDummy = self.dataDir + '/.snap/snap'
+
+        if self.workUnit:
+            assert self.snapEnable == False, \
+                self.logger.error('Workunit task and Snapshot enable cannot co-exist asserting rsync')
+        else:
+            # Create a data directory, sub directory and rsync directory
+            self.my_mnt.run_shell(["mkdir", "{}".format(self.dataDir)])
+            self.my_mnt.run_shell(["mkdir", "{}".format(self.sourceDir)])
+
+        # Create destination directory
+        self.my_mnt.run_shell(["mkdir", "rsyncdir"])
+        self.my_mnt.run_shell(["mkdir", "rsyncdir/dir1"])
+
+        while not self.stopping.is_set():
```

Comment:
Even i faced the problem but not immediately, it was at the end when workunit task completed and removed the directory and rsync was just inside while loop. tried increasing the waitTime at end of while loop not seen the problem.
may be i can change default waittime to 10s